keps/sig-node/5554-in-place-update-pod-resources-alongside-static-cpu-manager-policy/README.md

```diff
@@ -0,0 +1,1210 @@
+* Update allocateCPUs to:
+  + Pass the reusableCPUsForResize and mustKeepCPUsForResize to the policy takeByTopology methods and update the states.
+
+Updates in policy_static_test.go have been implemented.
+
+#### Introduced Error types for this KEP
+
+The following three Error InPlacePodVerticalScaling constants are introduced for this KEP
+
+```
+// InPlacePodVerticalScaling types
+const (
+    InconsistentCPUAllocationError = "inconsistentCPUAllocationError"
+    ProhibitedCPUAllocationError   = "prohibitedCPUAllocationError"
+    GetPromisedCPUSetError         = "getPromisedCPUSetError"
+)
+```
+
+#### Promised CPUs checkpoint
+
+As it was decided in [SIG Node meeting discussing static CPU policy support](https://www.youtube.com/watch?v=RuqzXH3liqg), 
+
+* CPUs allocated upon creation of Guaranteed PoDs need to be explicitly checkpointed locally in the worker node ( please refer to Static CPU Management Policy Support for more details ) we use the term “promised” CPUs in the rest of this document to refer to those CPUs.
+* Use of existing CPU Manager checkpoint should be used for storing the “promised” CPUs.
+
+To satisfy this requirement, CPUs allocated upon creation of Guaranteed PoDs need to be explicitly checkpointed, for this to happen CPU Manager State checkpoint file version 3 format need to be introduced in CPUManagerCheckpoint struct used to store cpu/pod assignments in a checkpoint v3 format.
+
+To ensure backwards compatibility with potential external applications using v2 format we propose the additional Promised variable to be of type map[string]map[string]string], where allocated CPUs upon creation for the first time of the Guaranteed PoD will be stored.
+
+```
+// CPUManagerCheckpoint struct is used to store cpu/pod assignments in a checkpoint in v3 format
+type CPUManagerCheckpoint struct {
+    PolicyName    string                       `json:"policyName"`
+    DefaultCPUSet string                       `json:"defaultCpuSet"`
+    Entries       map[string]map[string]string `json:"entries,omitempty"`
+    Promised      map[string]map[string]string `json:"promised,omitempty"`
+    Checksum      checksum.Checksum            `json:"checksum"`
+}
+```
+
+In addition to above the following list of functions must be added : 
+* migrateV1CheckpointToV3 and migrateV2CheckpointToV3 to ensure backwards compatibility and upgrade support
+* GetPromisedCPUSet and SetPromisedCPUSet
+* GetCPUPromised and SetCPUPromised
+
+#### API server and generation
+
+Not applicable.
+
+##### Client requests to update generation
+
+Not applicable.
+
+#### Kubelet and observedGeneration
+
+Not applicable.
+
+##### Mutable Fields Analysis
+
+Not applicable.
+
+#### Other writers of pod status
+
+Not applicable.
+
+#### Client requests to update observedGeneration
+
+Not applicable.
+
+#### Mirror pods
+
+Not applicable.
+
+#### Future enhancements
+
+To be determined.
```

Comment:
Add the [extended proposal by @Chunxia202410](https://docs.google.com/document/d/19-yzI41L6_XRj6l_27ylWSc114fzgP301FhxCPf0Dbg/edit?tab=t.0#heading=h.lk6ngyjqttpu) here

I'll defer to @ffromani on whether we want to make it a requirement for this KEP or not. IMO I am okay to leave it as an extended enhancement / future KEP, but at minimum we should add to the beta graduation requirements that we will reevaluate this decision